From fe2db7e11ec07904ebf5d9cc2a3d0b1a8ac89a82 Mon Sep 17 00:00:00 2001
From: Yurii Rashkovskii <me@yrashk.com>
Date: Wed, 14 Feb 2018 15:43:27 +0700
Subject: [PATCH] Problem: Web UI doesn't look great on mobile

1. Issue listing is two compressed horizontally
   (three issues per row)
2. Syntax-highlighted code goes beyond
   the boundary of the element

Solution: adjust styling to make it friendlier
for small screens.
---
 sit-web/webapp/src/dashboard-page/issue-style.html                | 8 +++++++-
 .../webapp/src/syntax-highlighting/syntax-highlighting-style.html | 1 +
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/sit-web/webapp/src/dashboard-page/issue-style.html b/sit-web/webapp/src/dashboard-page/issue-style.html
index 2633743..44c8c4e 100644
--- a/sit-web/webapp/src/dashboard-page/issue-style.html
+++ b/sit-web/webapp/src/dashboard-page/issue-style.html
@@ -2,7 +2,7 @@
     <template>
         <style>
             paper-card {
-                width: 20vw;
+                width: 100%;
                 margin: 10pt;
                 --paper-card-header-text: {
                     font-size: 14pt;
@@ -11,6 +11,12 @@
                 }
             }
 
+            @media only screen and (min-width: 768px) {
+                paper-card {
+                    width: 20vw;
+                }
+            }
+
             .open {
                 background-color: darkseagreen;
                 padding: 4pt;
diff --git a/sit-web/webapp/src/syntax-highlighting/syntax-highlighting-style.html b/sit-web/webapp/src/syntax-highlighting/syntax-highlighting-style.html
index 909e3f8..e983650 100644
--- a/sit-web/webapp/src/syntax-highlighting/syntax-highlighting-style.html
+++ b/sit-web/webapp/src/syntax-highlighting/syntax-highlighting-style.html
@@ -4,6 +4,7 @@
             #code {
                 background-color: #F0F0F0;
                 padding: 5pt;
+                overflow: auto;
             }
         </style>
     </template>
-- 
2.15.0

